evolver/tests/test_logging.py

```diff
@@ -24,14 +26,14 @@ def test_log_evolver_hookup():
     class HW(SensorDriver):
         def read(self):
             self.logger.warning("test from HW")
-            self.logger.log(EVENT, "event from HW", extra={"vial": 1})
+            self.logger.log(EVENT, "event from HW", extra=LogInfo(vial=1, event_class="something"))
```

Comment:
I am curious - what happens if I pass in extra and it's not an instance of LogInfo? would this part happen:
```
ValueError(f"Extra data must be json-serializable, got error: {e}")
```

Reply:
You could pass in extra alone, but if you wanted it to appear in history you would specifically do:

```
extra={'evolver_extra': {'stuff_for_history': ...}}
```

and you'd have to be sure `evolver_extra` was the same as `LogInfo.EXTRA_KEY`. The JSON check would still happen in the history hookup, but not at log time.